### PR TITLE
Replacing latest tag with stable for lightning-bolts links

### LINKED ISSUE #10427: Broken links in the Example section of the README
Hello all,

Was reading the README and realized that the links `Example` section of the [README](https://github.com/PyTorchLightning/pytorch-lightning#examples) are broken.

The fist MNIST hello world link works but the other links do not work.

Thanks,

Thomas

### PR DESCRIPTION
**Pull request by ThomVett**
Links to example tutorials use the latest tag:
- the tag works for pytorch-lightning links
- latest tag does not work for lightning-bolts links

Replacing `latest` with `0.4.0` for lightning-bolts links as they bring up the correct file.

Resolves #10427

### CODE REVIEW

**Review by SeanNaren — APPROVED** [review]
Thanks for this!

**Review by SeanNaren — CHANGES_REQUESTED** [review]
Thanks for the PR, a lot of these pieces moved into the deprecated portions of lightning-bolts. To make sure everything remains up to date, I think a lot of these can be found under:

https://lightning-bolts.readthedocs.io/en/latest/deprecated/

for example self_supervised is now https://lightning-bolts.readthedocs.io/en/latest/deprecated/models/self_supervised.html would you be able to update the links accordingly?

**Comment by ThomVett** [comment]
Works for me I'll update to the deprecated tags.

**Comment by ThomVett** [comment]
@SeanNaren updated the links, let me know if these work

**Review by akihironitta — COMMENTED** [review]
Confirmed that all the changed line are valid now :]

It seems like the following line is also outdated. Could you update it, too?
https://github.com/PyTorchLightning/pytorch-lightning/blame/f0a88107322677a060b002e46abd7e9085d43fd9/docs/source/extensions/callbacks.rst#L89

```diff
-    `bolts library <https://lightning-bolts.readthedocs.io/en/latest/callbacks.html>`_.
+    `bolts library <https://lightning-bolts.readthedocs.io/en/latest/>`_.
```

**Comment by SeanNaren** [comment]
Thanks @ThomVett :)

In a separate issue should revisit these links. Having them go to a deprecated section of any docs isn't nice haha. We have some amazing tutorial pages that I think we could use instead :) cc @tchaton

**Review by SeanNaren — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by akihironitta — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by tchaton** [comment]
@Borda

**Review by Borda — COMMENTED** [review]
still pls replace all latest by stable

**Review by InCogNiTo124 — APPROVED** [review]
Everything seems okay

**Comment by ThomVett** [comment]
sorry i did not get any notification about activity on this thread.

@akihironitta i have the impression that this line is automatically generated from sphinx or some auto doc generating software? not sure i should update the `.rst` file?

**Comment by akihironitta** [comment]
> @akihironitta i have the impression that this line is automatically generated from sphinx or some auto doc generating software? not sure i should update the .rst file?

@ThomVett I don't think so. It's written in the source at:
https://github.com/PyTorchLightning/pytorch-lightning/blame/master/docs/source/extensions/callbacks.rst#L89
```rst
    `bolts library <https://lightning-bolts.readthedocs.io/en/latest/callbacks.html>`_.
```

**Comment by akihironitta** [comment]
> still pls replace all latest by stable

@Borda If we replace `latest` with `stable`, the links will be 404 as the doc PRs in Bolts are not released yet. So, we need to either:

1. replace it with `stable` after including all docs update PRs into the stable branch or
2. keep `latest` as is.

**Comment by Borda** [comment]
@akihironitta thx, that is good finding, I may consider another way - make new Bolts release

**Comment by ThomVett** [comment]
Hello, the unit tests of this PR seem to fail in different stages, any advice on getting this PR merged? The documentation PR should not affect any tests / builds?

**Review by tchaton — APPROVED** [review]
LGTM !

File: README.md [hunk 1]
```diff
@@ -359,30 +359,30 @@ ______________________________________________________________________
 
 ###### Contrastive Learning
 
-- [BYOL](https://lightning-bolts.readthedocs.io/en/latest/self_supervised_models.html#byol)
-- [CPC v2](https://lightning-bolts.readthedocs.io/en/latest/self_supervised_models.html#cpc-v2)
-- [Moco v2](https://lightning-bolts.readthedocs.io/en/latest/self_supervised_models.html#moco-v2)
-- [SIMCLR](https://lightning-bolts.readthedocs.io/en/latest/self_supervised_models.html#simclr)
+- [BYOL](https://lightning-bolts.readthedocs.io/en/latest/deprecated/models/self_supervised.html#byol)
```
Comment (tchaton) [inline README.md:362]: @Borda Should we change `latest` to `stable` ?

Comment (Borda) [reply README.md:362]: rather always stable or specific tag

**Comment by akihironitta** [comment]
Hi @ThomVett, thank you for your contribution! As @\Borda suggested, could you replace `latest` with `stable`? We recently release a new version of Bolts, so the link should work now :)

**Review by akihironitta — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**